packages/devtools_app/lib/src/app.dart

```diff
@@ -284,21 +284,23 @@ class DevToolsAppState extends State<DevToolsApp> with AutoDisposeMixin {
               .where((p) => embed && page != null ? p.screenId == page : true)
               .where((p) => !hide.contains(p.screenId))
               .toList();
+          final connectedToFlutterApp =
+              serviceConnection.serviceManager.connectedApp?.isFlutterAppNow ??
+                  false;
           return MultiProvider(
             providers: _providedControllers(),
             child: DevToolsScaffold(
               embed: embed,
               page: page,
               screens: screens,
               actions: [
-                if (connectedToVmService)
-                  // TODO(https://github.com/flutter/devtools/issues/1941)
-                  if (serviceConnection
-                          .serviceManager.connectedApp?.isFlutterAppNow ??
-                      false) ...[
-                    const HotReloadButton(),
-                    const HotRestartButton(),
-                  ],
+                if (connectedToVmService) ...[
+                  // TODO(kenz): do we need to hide this for Dart Web?
+                  HotReloadButton(
+                    callOnVmServiceDirectly: !connectedToFlutterApp,
+                  ),
+                  const HotRestartButton(),
```

Comment:
Hot Restart isn't a concept in the Dart CLI apps, so this shouldn't be shown for non-Flutter applications.

Reply:
Yes, this button hides itself based on whether the service is available.

Reply:
added a comment to make that more clear